src/dapla_pseudo/v1/models.py

```diff
@@ -136,16 +86,105 @@ def keyset_list(self) -> t.Union[t.List[PseudoKeyset], None]:
         return None if self.keyset is None else [self.keyset]
 
 
+class PseudoFunctionKeywordArgs(BaseModel):
+    """Representation of the possible keyword arguments"""
+
+    key_id: PredefinedKeys
+    strategy: t.Optional[UnknownCharacterStrategy] = None  # Only used for FF31
+    version_timestamp: t.Optional[str] = None  # Only used for 'map-sid'
+
+    def __str__(self) -> str:
+        return ",".join(f"{k}={v}" for k, v in self.model_dump(by_alias=True).items() if v is not None)
+
+    model_config = ConfigDict(alias_generator=camelize, populate_by_name=True)
+
+
 class PseudoFunction(BaseModel):
     """Formal representation of a pseudo function.
 
-    Use to build up the string representation expected by pseudo service
+    Use to build up the string representation expected by pseudo service.
+
+    Syntax: "<function_type>(<kwarg_1>=x, <kwarg_2>=y)"
+
+    where <kwarg_1>, <kwarg_2>, etc. represents the keywords defined in PseudoFunctionKeywordArgs
     """
 
-    function_type: str
-    key: str
-    extra_kwargs: Optional[list[str]] = None
+    function_type: PseudoFunctionTypes
+    kwargs: PseudoFunctionKeywordArgs
 
     def __str__(self) -> str:
         """Create the function representation as expected by pseudo service."""
-        return f"{self.function_type}({', '.join([f'keyId={self.key}'] + (self.extra_kwargs or []))})"
+        return f"{self.function_type}({self.kwargs})"
+
+
+class PseudoFunctionRedact(BaseModel):
+    """Representation of the 'redact' function.
+
+    This function has sufficiently different representation than PseudoFunction such that it warrants its own model.
```

Comment:
I feel like this could fit in the `PseudoFunction` model if we just allowed specifying optional `args` as well, it'd be nice to keep the knowledge of how to build a rule in one place.